ts/screens/modal/UpdateAppModal.tsx

```diff
@@ -45,130 +46,105 @@ const styles = StyleSheet.create({
-    const cancelButtonProps = {
-      cancel: true,
-      block: true,
-      onPress: () => BackHandler.exitApp(),
-      title: I18n.t("global.buttons.close")
-    };
-    const updateButtonProps = {
-      block: true,
-      primary: true,
-      onPress: this.openAppStore,
-      title: I18n.t("btnUpdateApp")
-    };
-
-    return (
-      <FooterWithButtons
-        type="TwoButtonsInlineThird"
-        leftButton={cancelButtonProps}
-        rightButton={updateButtonProps}
-      />
-    );
-  }
+const UpdateAppModal: React.FC = () => {
+  // Disable Android back button
+  const handleBackPress = () => true;
 
-  // Different footer according to OS
-  get footer() {
-    return Platform.select({
-      ios: this.renderIosFooter(),
-      android: this.renderAndroidFooter()
-    });
-  }
+  useEffect(() => {
+    BackHandler.addEventListener("hardwareBackPress", handleBackPress);
 
-  public render() {
-    // Current version not supported
-    return (
-      <Modal>
-        <BaseScreenComponent
-          appLogo={true}
-          goBack={false}
-          accessibilityEvents={{ avoidNavigationEventsUsage: true }}
-        >
-          <Container>
-            <View style={styles.container}>
-              <H2>{I18n.t("titleUpdateApp")}</H2>
-              <Text style={styles.text}>{I18n.t("messageUpdateApp")}</Text>
-              <Image
-                style={styles.img}
-                source={require("../../../img/icons/update-icon.png")}
-              />
-              {this.state.hasError && (
-                <Text style={styles.textDanger}>
-                  {I18n.t("msgErrorUpdateApp")}
-                </Text>
-              )}
-            </View>
-          </Container>
-        </BaseScreenComponent>
-        {this.footer}
-      </Modal>
-    );
-  }
-}
+    return () =>
+      BackHandler.removeEventListener("hardwareBackPress", handleBackPress);
+  }, []);
+
+  // Reset the error state after a given timeout
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    if (error) {
+      setTimeout(() => setError(false), ERROR_MESSAGE_TIMEOUT);
+    }
+  }, [error]);
+
+  // Tries to open the native app store, falling to browser web store
+  const openAppStore = useCallback(async () => {
+    try {
+      await Linking.openURL(storeUrl);
+    } catch (e) {
+      try {
+        await Linking.openURL(webStoreURL);
+      } catch (e) {
+        setError(true);
+      }
+    }
+  }, [setError]);
+
+  return (
+    <Modal>
+      <BaseScreenComponent
+        appLogo={true}
+        goBack={false}
+        accessibilityEvents={{ avoidNavigationEventsUsage: true }}
+      >
+        <Container>
+          <View style={styles.container}>
+            <H2>{I18n.t("titleUpdateApp")}</H2>
+            <Text style={styles.text}>{I18n.t("messageUpdateApp")}</Text>
+            <Image
+              style={styles.img}
+              source={require("../../../img/icons/update-icon.png")}
```

Comment:
In order to use the same pattern of all the codebase move this to the top of the component and use the import pattern:
`import imageName from "../../../img/icons/update-icon.png"`